### ISSUE TITLE
SPR001 false positives for two-argument super()

### ISSUE THREAD

**Issue by andersk** [charliermarsh/ruff#316]
`super(MyClass, arg)` can only be replaced with `super()` if `arg` is the first argument to a function that’s defined within `class MyClass`. In other cases, the `SPR001` diagnostic from ruff (and from flake8-super, I guess) is a false positive.

Some examples:

```python
class BaseClass:
    def f(self):
        pass


def defined_outside(self):
    super(MyClass, self).f()  # CANNOT use super()


class MyClass(BaseClass):
    def normal(self):
        super(MyClass, self).f()  # can use super()
        super().f()

    def different_argument(self, other):
        super(MyClass, other).f()  # CANNOT use super()

    def comprehension_scope(self):
        [super(MyClass, self).f() for x in [1]]  # CANNOT use super()

    def inner_functions(self):
        def outer_argument():
            super(MyClass, self).f()  # CANNOT use super()

        def inner_argument(self):
            super(MyClass, self).f()  # can use super()
            super().f()

        outer_argument()
        inner_argument(self)

    def inner_class(self):
        class InnerClass:
            super(MyClass, self).f()  # CANNOT use super()

            def method(inner_self):
                super(MyClass, self).f()  # CANNOT use super()

        InnerClass().method()

    defined_outside = defined_outside


MyClass().normal()
MyClass().different_argument(MyClass())
MyClass().comprehension_scope()
MyClass().inner_functions()
MyClass().inner_class()
MyClass().defined_outside()
```

Some false positives in the wild:

https://github.com/zulip/zulip/blob/5.6/zerver/lib/test_helpers.py#L167
https://github.com/zulip/zulip/blob/5.6/zerver/tests/test_auth_backends.py#L326

**Comment by charliermarsh** [comment]
Nice, good call, will fix.

**Comment by charliermarsh** [comment]
Working on this now.
